### PR TITLE
Replace request with war in uma.js

### LINKED ISSUE #205: Update uma.js to use got lib and remove request
https://github.com/GluuFederation/gluu-passport/issues/204

### PR DESCRIPTION
**Pull request by kdhttps**
close #205

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,8 +58,6 @@
     "prom-client": "^12.0.0",
     "ramda": "^0.27.1",
     "redis": "^3.0.2",
-    "request": "^2.88.0",
```
Comment (christian-hawk) [inline package.json:61]: `got` is still in `devDependencies` object. Shouldn't it be in `dependencies' ?

File: server/utils/uma.js [hunk 2]
```diff
@@ -16,24 +17,30 @@ let rpt
...
+      const body = response.body
       logger.log2(
         'debug',
-        `getTokenEndpoint. obj = ${JSON.stringify(obj)}`
+        `getTokenEndpoint. obj = ${JSON.stringify(body)}`
       )
-      const endpoint = obj.token_endpoint
-      if (endpoint) {
+      const tokenEndpoint = body.token_endpoint
+      if (tokenEndpoint) {
```
Comment (christian-hawk) [inline server/utils/uma.js:28]: `tokenEndpoint`! Nice, clean coding 👨🏻‍🎤

File: server/utils/uma.js [hunk 3]
```diff
@@ -104,29 +116,29 @@ function doRequest (options) {
...
             'getConfiguration. Got www-authenticate in header' +
           `with ticket ${parsed.parms.ticket}`)
           logger.log2('debug',
           `getConfiguration. Reponse Headers ${JSON.stringify(
-            response.headers, null, 4)}`
+            headers, null, 4)}`
           )
           logger.log2('debug',
-          `getConfiguration. parsed.params = ${parsed.params}`
+          `getConfiguration. parsed.params = ${JSON.stringify(parsed.parms)}`
```
Comment (christian-hawk) [inline server/utils/uma.js:141]: should it be `parsed.parms`, not `parsed.params`? 
If this is wrong, a test should break it. If not, it's all good.

File: test/configDiscovery.test.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+/* eslint-disable security/detect-non-literal-fs-filename */
```
Comment (christian-hawk) [inline test/configDiscovery.test.js:1]: you are getting better on it.

**Review by christian-hawk — CHANGES_REQUESTED** [review]
I'm impressed with your testing and good practices evolution.

Please kindly check my request comments.

File: package.json (hunk 1 above)
Comment (kdhttps) [reply package.json:61]: Right, I'll add it in `dependencies`.

File: server/utils/uma.js (hunk 3 above)
Comment (kdhttps) [reply server/utils/uma.js:141]: `parsed.parms` is correct one.

File: test/configDiscovery.test.js (hunk 4 above)
Comment (kdhttps) [reply test/configDiscovery.test.js:1]: thanks mate :)

**Review by christian-hawk — APPROVED** [review]

**Pull request merged**